### DISCUSSION TITLE
Handling Edge functions streaming connection reset/abort

### DISCUSSION THREAD

**Discussion by dqunbp** [vercel/next.js#48381] [Help]
### Summary

Hi! I used Edge functions [Streaming functionality](https://vercel.com/docs/concepts/functions/edge-functions/streaming).
When I abort the stream on the client side, I have a such error in the edge function logs:

```
error - uncaughtException: Error: aborted
    at connResetException (node:internal/errors:711:14)
    at abortIncoming (node:_http_server:743:17)
    at socketOnClose (node:_http_server:737:3)
    at Socket.emit (node:events:525:35)
    at TCP.<anonymous> (node:net:298:12) {
  code: 'ECONNRESET'
}
```

How can I handle this on the server side?

As a server-side example, I used the edge function from this repo
https://github.com/Nutlope/twitterbio/blob/main/pages/api/generate.ts

Then I send the query from the Postman, and canceled the response before it finished

**Comment by enricoros** [comment]
+1 Very important in this day and age of LLMs trickling down bytes.

**Reply by siddarthvader** [reply]
any update on this ? @dqunbp

**Comment by flosrn** [comment]
did you find a solution?

**Comment by railty** [comment]
Same problem 13.5.6

**Reply by laflechaenelaire** [reply]
Same, 14.0

**Comment by brycedbjork** [comment]
Same problem

**Comment by shredEngineer** [comment]
Any solution? This is absolutely crucial! I need to bill my customers for LLM tokens used, even when aborting the stream. This needs to be handled.

**Comment by mortezamehrabi** [comment]
same problem too!
is there any solution?
